src/libstore/globals.hh

```diff
@@ -975,6 +975,19 @@ public:
           resolves to a different location from that of the build machine. You
           can enable this setting if you are sure you're not going to do that.
         )"};
+
+    Setting<bool> useXDGBaseDirectories{
+        this, false, "use-xdg-base-directories",
+        R"(
+          If set to `true`, Nix will conform to the [XDG Base Directory Specification]
+          for files in $HOME.
+
+          Note that this changes the location of some well-known symlinks that Nix
+          creates, which might cause some breakage.
+
```

Comment:
This part is the actually important one. It should document all the paths used which fall in the XDG domain, old and new. All the other occurrences should link here.